cluster.go

```diff
@@ -55,8 +65,20 @@ func newClusterClient(opt *ClientOption, connFn connFn, retryer retryHandler) (*
 	if opt.ReplicaOnly && opt.SendToReplicas != nil {
 		return nil, ErrReplicaOnlyConflict
 	}
+	if opt.ReplicaOnly && opt.ReaderNodeSelector != nil {
+		return nil, ErrReplicaOnlyConflictWithReaderNodeSelector
+	}
+	if opt.SendToReplicas != nil && opt.ReaderNodeSelector != nil {
+		return nil, ErrSendToReplicasConflictWithReaderNodeSelector
```

Comment:
I feel that we should not hide `SendToReplicas` when setting `ReaderNodeSelector`. That looks too implicit. Actually, I think we should ask users to provide  `SendToReplicas` when `ReaderNodeSelector` is set:

Instead of simply (ReaderNodeSelector alone is hard for me to reason what it actually does):
```go
	client, err := rueidis.NewClient(rueidis.ClientOption{
		ReaderNodeSelector: func(slot uint16, replicas []rueidis.ReplicaInfo) int {
			return rand.IntN(len(replicas))
		},
	})
```

I prefer making it more explicitly state:
```go
	client, err := rueidis.NewClient(rueidis.ClientOption{
		SendToReplicas: func(cmd rueidis.Completed) bool {
			return cmd.IsReadOnly()
		},
		ReplicaSelector: func(slot uint16, replicas []rueidis.ReplicaInfo) int {
			return rand.IntN(len(replicas))
		},
	})
```
I believe the latter provides users with a better sense of control over which replicas to execute which commands.

Also, I think changing the name of ReaderNodeSelector to ReplicaSelector is more consistent with the existing naming.

Reply:
Ok, Using `SendToReplicas` and `ReplicaSelector` both makes sense to me. Even though user can select primary using `ReplicaSelector`, It is configured explicitly, so it is clear to me. 
[90387d2](https://github.com/redis/rueidis/pull/692/commits/90387d2a94e31990bb38b2b5e0d0bf68123db9d7)